Source/Dafny/Dafny.atg

```diff
@@ -4061,13 +4080,19 @@ CasePattern<.out CasePattern<BoundVar> pat.>
     ")"                                (. pat = new CasePattern<BoundVar>(id, id.val, arguments); .)
   | "("                                (. id = t;
                                           arguments = new List<CasePattern<BoundVar>>();
+                                          ghostArguments = new List<bool>();
                                        .)
-      [ CasePattern<out pat>           (. arguments.Add(pat); .)
-        { "," CasePattern<out pat>     (. arguments.Add(pat); .)
+      [ [ "ghost"                      (. isGhost = true; .)
+        ]
+        CasePattern<out pat>           (. arguments.Add(pat); ghostArguments.Add(isGhost); .)
+        { ","                          (. isGhost = false; .)
+          [ "ghost"                    (. isGhost = true; .)
```

Comment:
Does it mean that it's fine to call `theBuiltIns.TupleType(id, arguments.Count, true, ghostArguments);` with the number of arguments instead of the number of ghost arguments, or should I try to remove it? I don't understand why the parser has to emit these tuple types. I expect the types to be fully known only after type checking rather than at parsing time.

Reply:
Removing the `theBuiltIns.TupleType(id, arguments.Count, true, ...);` call doesn't seem to cause big problems in the test suite. Just a few tests fail because of unexpected C++ warnings, but they might be caused by other reasons. I'm debugging.